ecommerce/views.py

```diff
@@ -148,29 +149,36 @@ def post(self, request, *args, **kwargs):
         receipt.order = order
         receipt.save()
 
-        if request.data['decision'] != 'ACCEPT':
-            # This may happen if the user clicks 'Cancel Order'
+        decision = request.data['decision']
+        if order.status == Order.FAILED and decision == 'CANCEL':
+            # This is a duplicate message, ignore since it's already handled
+            return Response()
+        elif order.status != Order.CREATED:
+            raise EcommerceException("Order {} is expected to have status 'created'".format(order.id))
+
+        if decision != 'ACCEPT':
```

Comment:
is it reasonable to move all the CyberSource statuses in a proper class just to have them centralized? Feel free to postpone.